### PR TITLE
Add different default sampling rates for when RC fetch failed

### PR DESCRIPTION
**Pull request by mrober**

### CODE REVIEW

File: firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigResolver.java [hunk 1]
```diff
@@ -331,7 +338,12 @@ public double getNetworkRequestSamplingRate() {
       return deviceCacheValue.get();
     }
 
-    // 3. Returns default value if there is no valid value from above approaches.
+    // 3. Check if RC fetch failed.
+    if (remoteConfigManager.isLastFetchFailed()) {
```
Comment (visumickey) [inline firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigResolver.java:342]: What is the meaning of lastFetchFailed? Are we accounting for any kind of a fetch failure to fallback to these defaults?

File: firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java [hunk 2]
```diff
@@ -218,6 +230,11 @@ protected Long getDefault() {
       return 100L;
     }
 
+    @Override
+    protected Long getDefaultOnRcFetchFail() {
+      return getDefault() * 3;
```
Comment (visumickey) [inline firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java:235]: Let's add a comment to say that we are increasing the default 3 folds. Eg: 100ms to 300ms.

File: firebase-perf/src/test/java/com/google/firebase/perf/config/ConfigurationConstantsTest.java [hunk 3]
```diff
@@ -91,6 +91,7 @@ public void getInstance_SessionsCpuCaptureFrequencyForegroundMs_validateConstant
         SessionsCpuCaptureFrequencyForegroundMs.getInstance();
 
     assertThat(configFlag.getDefault()).isEqualTo(100L);
+    assertThat(configFlag.getDefaultOnRcFetchFail()).isEqualTo(300L);
```
Comment (visumickey) [inline firebase-perf/src/test/java/com/google/firebase/perf/config/ConfigurationConstantsTest.java:94]: Would we have such asserts for the sampling rates?

**Review by visumickey — APPROVED** [review]
Thank you for making those changes.

File: firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java (hunk 2 above)
Comment (mrober) [reply firebase-perf/src/main/java/com/google/firebase/perf/config/ConfigurationConstants.java:235]: Done

File: firebase-perf/src/test/java/com/google/firebase/perf/config/ConfigurationConstantsTest.java (hunk 3 above)
Comment (mrober) [reply firebase-perf/src/test/java/com/google/firebase/perf/config/ConfigurationConstantsTest.java:94]: Added.

**Pull request merged**